x/distribution/keeper/store.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+)
+
+// get the global fee pool distribution info
+func (k Keeper) GetFeePool(ctx sdk.Context) (feePool types.FeePool) {
+	store := ctx.KVStore(k.storeKey)
+	b := store.Get(FeePoolKey)
+	if b == nil {
+		panic("Stored fee pool should not have been nil")
+	}
+	k.cdc.MustUnmarshalBinaryLengthPrefixed(b, &feePool)
```

Comment:
These should all not be length prefixed, right?

Reply:
Without length prefixing, empty structs serialize to `nil`, which we cannot set in the store.

Reply:
but why would nil structs go in the store, theres a `b == nil` check right above this? Also nil things should be able to go into the store, we should adapt API's so that returning `nil` is different from there being no element there. (As you can have a 0 byte element there, that is a valid leaf in the tree)

Reply:
They'll often go in the store, e.g. for zero-coins, which is an empty struct.

I agree about changing the APIs - https://github.com/cosmos/cosmos-sdk/issues/3274